daemon/src/maker_cfd.rs

```diff
@@ -630,16 +698,15 @@ impl Actor {
 
         let taker_id = self.get_taker_id_of_proposal(&order_id)?;
 
-        // TODO: Initiate the settlement - should we start calculating the
-        // signature here?
-
         self.takers
             .do_send_async(maker_inc_connections::TakerMessage {
                 taker_id,
                 command: TakerCommand::NotifySettlementAccepted { id: order_id },
             })
             .await?;
 
+        self.current_agreed_proposals
+            .insert(order_id, self.get_settlement_proposal(order_id)?);
         self.remove_pending_proposal(&order_id)
             .context("accepted settlement")?;
```

Comment:
Removing the proposal from the hashmap would return you the owned instance in case it is present. I think this would be cleaner if we would just:

1. Always try to remove it
2. Fail if it wasn't there
3. If it was there, move it into the other `HashMap`.

Reply:
Alternatively, we could have a single `HashMap` with a data type that can internally transition from Proposed -> Accepted.